src/main/java/uk/ac/ebi/ampt2d/metadata/rest/controllers/AssemblyRestController.java

```diff
@@ -51,8 +59,12 @@
     })
     @RequestMapping(method = RequestMethod.GET, path = "search", produces = "application/json")
     @ResponseBody
-    public Resources<Assembly> search(@QuerydslPredicate(root = Assembly.class) Predicate predicate) {
-        return new Resources<>(assemblyRepository.findAll(predicate));
+    public ResponseEntity<Resources<?>> search(@QuerydslPredicate(root = Assembly.class) Predicate predicate) {
+        List<Assembly> assemblies = (List<Assembly>) assemblyRepository.findAll(predicate);
+
+        Resources<?> resources = resourceAssembler.toResources(Assembly.class, assemblies);
+
```

Comment:
can u please cast and return to Resources<AssemblyResource> so that the response model shows assembly entity instead of empty content.